File: Programmers/Lv0/코드 처리하기.py
# 내 풀이
def solution(code):
    mode = 0
    result = []
    if len(code) == 0:
        return "EMPTY"

    for x in range(len(code)):
        if code[x] == "1":
            if mode == 0:
                mode = 1
                continue
            elif mode == 1:
                mode = 0
                continue

        else:
            if mode == 0 and x % 2 == 0:
                result.append(code[x])

            elif mode == 1 and x % 2 == 1:
                result.append(code[x])

            else:
                continue

    if len(result) == 0:
        return "EMPTY"

    return "".join(result)
# 다른 사람 풀이

def solution(code):
    return "".join(code.split("1"))[::2] or "EMPTY"

File: Programmer_Java/Lv0/코드 처리하기.java
// package Programmers;

// public class 코드 처리하기 {
  
// }

// 출처: 프로그래머스
// https://school.programmers.co.kr/learn/courses/30/lessons/181932

// 풀이과정_ 개선 중 
/*
 * class Solution {
    public String solution(String code) {
        // 시작 할 떄 모드 
        int mode = 0;
        int num = code.length();
        String result = "";
        for(int i=0; i < num; i++){
            if(code.charAt(i)=='1'){
                if(mode==1){
                    mode = 0;
                }
                else{
                    mode = 1; 
                }
            }
            
            else{
                if(mode==0){
                    if(i%2==0){
                        result+=code.charAt(i);
                    }
                }
                else{
                    if(i%2==1){
                        result+=code.charAt(i);
                    }
                }
            }
        }
        String answer = "";
        return result;
    }
}
 */